typegen/src/typegen/settings/substitutes.rs

```diff
@@ -377,6 +350,31 @@ pub fn absolute_path(path: syn::Path) -> Result<AbsolutePath, TypeSubstitutionEr
     path.try_into()
 }
 
+/// New-type trait for `TryInto<syn::Path>`
+pub trait TryIntoSynPath {
+    /// Turns the type into a [`syn::Path`]. Returns None for empty paths.
+    fn syn_path(self) -> Option<syn::Path>;
+}
```

Comment:
This is cleaner than the `PathSegments` stuff I think :)

Reply:
Yeah, I thought about that too and realised that you had no other choice! The other half way option would have been to define an `enum Path { SynPath(syn::Path), TypeInfoPath(typeinfo::Path) }` and then a `From` on that, but I think your approach is cleaner since you only care about getting one of those back!